### PR TITLE
completed typescript-forms-Carolyne Golias

### PR DESCRIPTION
**Pull request by GCarolyne**
Screenshot 2024-11-07 at 12 14 25 PM
Screenshot 2024-11-07 at 12 15 03 PM

### CODE REVIEW

File: typescript-forms/form-events.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+if (!$input1) throw new Error('the $input1 query failed');
+const $textAr = document.querySelector('textarea');
+if (!$textAr) throw new Error('the $textAr query failed');
+
+function handleFocus(event: Event): void {
+  console.log('focus event fired');
+  const eventTarget = event.target as HTMLInputElement | HTMLTextAreaElement;
+  console.log('eventTarget.name', eventTarget.name);
+}
+$input.addEventListener('focus', handleFocus);
```
Comment (shawnkost) [inline typescript-forms/form-events.ts:13]: `$input` should also get a `blur` event and an `input` event

File: typescript-forms/form-events.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+function handleBlur(event: Event): void {
+  console.log('blur event fired');
+  const eventTarget = event.target as HTMLInputElement | HTMLTextAreaElement;
+  console.log('eventTarget.name', eventTarget.name);
+}
+$input1.addEventListener('blur', handleBlur);
```
Comment (shawnkost) [inline typescript-forms/form-events.ts:22]: `$input1` should also get a `focus` event and an `input` event

File: typescript-forms/form-events.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+function handleInput(event: Event): void {
+  const eventTarget = event.target as HTMLInputElement | HTMLTextAreaElement;
+  console.log('eventTarget.name', eventTarget.name);
+  console.log('eventTarget.value', eventTarget.value);
+}
+
+$textAr.addEventListener('input', handleInput);
```
Comment (shawnkost) [inline typescript-forms/form-events.ts:32]: `$textAr` should also get a `focus` event and an `blur` event

**Review by shawnkost — CHANGES_REQUESTED** [review]
Just a few small changes here Carolyne!

**Comment by GCarolyne** [comment]
Screenshot 2024-11-18 at 10 09 51 AM

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**